src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -224,7 +230,11 @@ export class Kernel implements IKernel {
             );
         }
     }
-    private async startNotebook(options: { disableUI?: boolean; document: NotebookDocument }): Promise<INotebook> {
+    private async startNotebook(options: { disableUI: boolean }): Promise<INotebook> {
+        if (!options.disableUI) {
+            // This means the user is actually running something against the kernel (deliberately).
+            initializeNotebookTelemetryBasedOnUserAction(this.notebookDocument.uri, this.kernelConnectionMetadata);
```

Comment:
If we're displaying a UI, then we know the user actually started the notebook deliberately.
Meaning the user ran a cell.